qutebrowser/misc/sessions.py

```diff
@@ -387,9 +391,14 @@ def _load_tab(self, new_tab, data):
                     histentry['original-url'].encode('ascii'))
             else:
                 orig_url = url
+            if histentry.get("last_visited"):
+                last_visited = QDateTime.fromString(histentry.get("last_visited"))
+            else:
+                last_visited = None
             entry = TabHistoryItem(url=url, original_url=orig_url,
                                    title=histentry['title'], active=active,
-                                   user_data=user_data)
+                                   user_data=user_data,
+                                   last_visited=last_visited)
```

Comment:
Does this actually work as-is? Wouldn't we need to find out how that date is (de)serialized in `tabhistory.py` ([webkit](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/browser/webkit/tabhistory.py), [webengine](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/browser/webengine/tabhistory.py)) for this to work?

Reply:
It was not working. Chromium wants the timestamp as the number of microseconds since 1601-01-01. So [this](https://github.com/qutebrowser/qutebrowser/blob/6f028e9/qutebrowser/browser/webengine/tabhistory.py#L63) would have always been in January 1601.

Reply:
And `QWebHistoryItem::lastVisited()` appears to always return `QDateTime()`... It's currently falling back to 1970-01-01 for invalid (and earlier than 1970) dates so I think I will just leave that as is for webkit. [Here](https://bugs.webkit.org/show_bug.cgi?id=60940) is an old bug (marked resolved invalid) that says something similar (presumably with a local timezone adjustment).